examples/julia/additional_argument_test.jl

```diff
@@ -11,14 +11,14 @@ end
 # This handler expects an additional argument `received_timestamps`. To
 # ensure that argument is provided, we just have to pass it as an
 # additional argument to `subscribe()`
-function handler(channel::String, msg::example_t, received_timestamps::Vector)
+function handler(rbuf, channel::String, msg::example_t, received_timestamps::Vector)
     println("Received message on channel: ", channel)
     @assert msg.timestamp == length(received_timestamps) "Received message with incorrect timestamp"
     push!(received_timestamps, msg.timestamp)
 end
 
 received_timestamps = Int[]
-sub = subscribe(zcm, "EXAMPLE", example_t, handler, received_timestamps)
+sub = subscribe(zcm, "EXAMPLE", typed_handler(handler, example_t), received_timestamps)
```

Comment:
Any way to overload subscribe so we don't have to have this typed_handler thing as the only way to subscribe?
Syntax like `subscribe<example_t>(zcm, "EXAMPLE", handler)` would be pretty awesome

Reply:
Alas, prefixing with a type like that isn't possible for functions, but there are a lot of other options. We just have to decide what the right set of signatures is. Here's one option:

```julia
# untyped subscription
subscribe(zcm, channel, handler)

# typed subscription 
subscribe(zcm, channel, handler, msgtype)

# typed subscription with extra args
subscribe(zcm, channel, handler, msgtype, extra1, extra2)

# untyped subscription with extra args
subscribe(zcm, channel, handler, Void, extra1, extra2)
```

Reply:
Yea I like that suggestion a lot. Optimize the syntax for the main use case (which will almost definitely be typed subscriptions). It also mirrors nodejs a little bit more too in how you pass in Void (None) if you want it untyped.

Reply:
Ok, see 130d253